pkg/plugin/processor/builtin/webhook_http.go

```diff
@@ -0,0 +1,309 @@
+
+package builtin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"maps"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/jpillora/backoff"
+	"github.com/mitchellh/mapstructure"
+)
+
+var defaultWebhookHTTPConfig = map[string]string{
+	"request.method":      http.MethodPost,
+	"request.contentType": "application/json",
+
+	"backoffRetry.count":  "0",
+	"backoffRetry.min":    "100ms",
+	"backoffRetry.max":    "5s",
+	"backoffRetry.factor": "2",
+
+	"request.body":  ".",
+	"response.body": ".Payload.After",
+}
+
+type webhookHTTPConfig struct {
+	URL         string `mapstructure:"request.url"`
+	Method      string `mapstructure:"request.method"`
+	ContentType string `mapstructure:"request.contentType"`
+
+	BackoffRetryCount  float64       `mapstructure:"backoffRetry.count"`
+	BackoffRetryMin    time.Duration `mapstructure:"backoffRetry.min"`
+	BackoffRetryMax    time.Duration `mapstructure:"backoffRetry.max"`
+	BackoffRetryFactor float64       `mapstructure:"backoffRetry.factor"`
+
+	// RequestBodyRef specifies which field from the input record
+	// should be used as the body in the HTTP request.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	RequestBodyRef *sdk.ReferenceResolver `mapstructure:"request.body"`
+	// ResponseBodyRef specifies to which field should the
+	// response body be saved to.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	ResponseBodyRef *sdk.ReferenceResolver `mapstructure:"response.body"`
+	// ResponseStatusRef specifies to which field should the
+	// response status be saved to.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	ResponseStatusRef *sdk.ReferenceResolver `mapstructure:"response.status"`
+}
+
+type webhookHTTP struct {
+	sdk.UnimplementedProcessor
+
+	logger log.CtxLogger
+
+	config webhookHTTPConfig
+}
+
+func NewWebhookHTTP(l log.CtxLogger) sdk.Processor {
+	return &webhookHTTP{logger: l.WithComponent("builtin.webhookHTTP")}
+}
+
+func (w *webhookHTTP) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "webhook.http",
+		Summary: "HTTP webhook processor",
+		Description: `
+A processor that sends an HTTP request to the specified URL with the specified HTTP method
+(default is POST) with a content-type header as the specified value (default is application/json).
+
+The whole record as json will be used as the request body and the raw response body will be set under 
+Record.Payload.After. If the response code is (204 No Content) then the record will be filtered out.
+`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: nil,
```

Comment:
Waiting for #1289 to finish this.

Reply:
You got it, the code is in the feature branch. Also you can and should now use [paramgen](https://github.com/ConduitIO/conduit-commons/tree/main/paramgen) to make your life simpler 🙂